core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchFragment.kt

```diff
@@ -150,7 +150,9 @@ class SearchFragment : BaseFragment() {
     searchMenuItem.expandActionView()
     searchView = searchMenuItem.actionView as SearchView
     searchView.setOnQueryTextListener(SimpleTextListener {
-      searchViewModel.actions.offer(Filter(it))
+      if (it.isNotEmpty()) {
+        searchViewModel.actions.offer(Filter(it))
+      }
```

Comment:
This seems needless to me, how it can be empty when we're initializing and assigning every param before? Can you even justify it?

Reply:
No, @s-ayush2903 Initially I thought so it was redundant. After some debugging that `OnQueryTextListener` s `onQueryTextChange` initially gives some empty String.
```
searchViewModel.actions.offer(Filter(it))
```
Since the above line code is time-consuming. then it will be redundant or power-consuming to search for an empty string. If you want then I'll give you some log.